src/DataProtection/DataProtection/src/MetricsLoggerProvider.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Metrics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Xml.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Win32;
+
+namespace Microsoft.AspNetCore.DataProtection;
+
+internal sealed class MetricsLoggerProvider : ILoggerProvider
+{
+    public const string MeterName = "Microsoft.AspNetCore.DataProtection";
+
+    private readonly Meter? _meter;
+
+    private readonly ILogger _logger;
+
+    public MetricsLoggerProvider()
+    {
+        _logger = NullLogger.Instance;
+    }
+
+    public MetricsLoggerProvider(IMeterFactory meterFactory)
+    {
+        _meter = meterFactory.Create(MeterName);
+
+        var errorCounter = _meter.CreateCounter<long>(
+            "errors",
+            unit: "{error}",
+            description: "Number of errors that have occurred.");
+        var warningCounter = _meter.CreateCounter<long>(
+            "warnings",
+            unit: "{warning}",
+            description: "Number of warnings that have occurred.");
+
+        _logger = new MetricsLogger(errorCounter, warningCounter);
+    }
+
+    ILogger ILoggerProvider.CreateLogger(string _categoryName) => _logger;
+
+    void IDisposable.Dispose() => _meter?.Dispose();
+
+    private sealed class MetricsLogger : ILogger
+    {
+        private readonly Counter<long> _errorCounter;
+        private readonly Counter<long> _warningCounter;
```

Comment:
I think this would be one counter and the loglevel is a tag.

Reply:
Sure.  Is there a doc somewhere on tag naming conventions?

Reply:
Recommendations are here: https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics-instrumentation#best-practices-4

Existing metrics that you can look at for prior art examples: https://learn.microsoft.com/en-us/dotnet/core/diagnostics/built-in-metrics

Reply:
I found this registry of known attributes: https://opentelemetry.io/docs/specs/semconv/attributes-registry/